src/binary-search-tree.js

```diff
@@ -5,6 +5,31 @@ class BinarySearchTree {
     this.left = null;
     this.right = null;
   }
+  insert(value) {
+    const node = new BinarySearchTree(value);
+    const recFunction = (obj) => {
+      if (value < obj.value && obj.left === null) obj.left = node;
+      else if (value < obj.value) recFunction(obj.left);
+      else if (value > obj.value && obj.right === null) obj.right = node;
+      else if (value > obj.value) recFunction(obj.right);
+    };
+    recFunction(this);
+  }
+  contains(value) {
```

Comment:
Here, your recursive function never returns, so it will end up blowing the stack. Your base case needs to specify when the recursive function returns. Additionally, you're going to want to return as soon as the value you're looking for is found, instead of just setting a boolean, because your contains function will then keep running because you never told it to return. This means it's doing unnecessary work that doesn't need to be done since you've already found the value you're looking for, but you didn't tell your function to end its execution.